### PR TITLE
fix decorator checks in plugin definitions

### PR DESCRIPTION
**Pull request by mcollina**
Fixes https://github.com/fastify/fastify/issues/3332

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by SimenB — APPROVED** [review]
thanks! can confirm it works in my real project as well 🙂

**Review by jsumners — APPROVED** [review]
Looks good to me. Please issue patch release.

File: test/decorator.test.js [hunk 1]
```diff
@@ -904,3 +904,38 @@ test('Request/reply decorators should be able to access the server instance', as
     t.equal(this.server.foo, 'bar')
   }
 })
+
+test('plugin required decorators', async t => {
```
Comment (Eomm) [inline test/decorator.test.js:908]: Could we add a test for the failure case?

I see only one that checks the `The decorator '${decorator}'${withPluginName} is not present in ${instance}` case

Comment (mcollina) [reply test/decorator.test.js:908]: That is already covered by current unit test

**Pull request merged**